WooCommerce/Classes/Authentication/WPComLogin/WPComEmailLoginViewModel.swift

```diff
@@ -78,8 +87,16 @@ final class WPComEmailLoginViewModel: ObservableObject {
             }
             await startAuthentication(email: email, isPasswordlessAccount: passwordless)
         } catch {
-            analytics.track(event: .JetpackSetup.loginFlow(step: .emailAddress, failure: error))
-            onError(error.localizedDescription)
+            if allowAccountCreation,
```

Comment:
wdyt about replacing the if-else with guard?

```
guard allowAccountCreation,
      let apiError = error as? WordPressAPIError<WordPressComRestApiEndpointError>,
      case .endpointError(let endpointError) = apiError,
      endpointError.apiErrorCode == Constants.unknownUserErrorCode else {
    analytics.track(event: .JetpackSetup.loginFlow(step: .emailAddress, failure: error))
    onError(error.localizedDescription)
    return
}

await requestAuthenticationLink(email: email, forAccountCreation: true)
```

I think `guard` helps with early return pattern and communicates what conditions are required, and reduces nesting a bit.

Reply:
Applied in [99ad008](https://github.com/woocommerce/woocommerce-ios/pull/14431/commits/99ad008f9269d0b2f94e423bcaeffd6118e2e60f)